youtube_dl/extractor/cda.py

```diff
@@ -38,32 +48,71 @@ class CDAIE(InfoExtractor):
     }]
 
     def _real_extract(self, url):
+        UPLOADER_REGEX = r'''(?x)
+            <span[^>]*itemprop=(["\'])author\1[^>]*>
+            (?:<span[^>]*>[^<]*</span>|(?!</span>)(?:.|\n))*
+            <(span|meta)[^>]*itemprop=(["\'])name\3[^>]*>(?P<uploader>[^<]+)</\2>[^<]*</span>
+        '''
+        RATING_REGEX = r'''(?x)
+            <span[^>]*itemprop=(["\'])aggregateRating\1[^>]*>
+            (?:<span[^>]*>[^<]*</span>|(?!</span>)(?:.|\n))*
+            <(span|meta)[^>]*itemprop=(["\'])ratingValue\3[^>]*>(?P<rating_value>[0-9.]+)[^<]*
+            <(span|meta)[^>]*itemprop=(["\'])bestRating\6[^>]*>(?P<best_rating>[0-9.]+)</\5>[^<]*</\2>
+        '''
+
         video_id = self._match_id(url)
-        webpage = self._download_webpage('http://ebd.cda.pl/0x0/' + video_id, video_id)
+        self._set_cookie('cda.pl', 'cda.player', 'html5')
+        webpage = self._download_webpage(
+            self._BASE_URL + '/video/' + video_id, video_id)
 
         if 'Ten film jest dostępny dla użytkowników premium' in webpage:
-            raise ExtractorError('This video is only available for premium users.', expected=True)
-
-        title = self._html_search_regex(r'<title>(.+?)</title>', webpage, 'title')
+            raise ExtractorError(
+                'This video is only available for premium users.',
+                expected=True)
 
         formats = []
 
+        # Set uploader to None if not found.
+        # There are many videos without uploader specified.
+        uploader = self._search_regex(UPLOADER_REGEX, webpage, 'uploader',
+                                      default=None, group='uploader')
+
+        # Set view_count to None if not found.
+        # There are many older videos without view counter available.
+        view_count = self._search_regex(
+            r'Odsłony:(?:\s|&nbsp;)*([0-9]+)', webpage, 'view_count', default=None)
+
+        m_rating = re.search(RATING_REGEX, webpage)
+        average_rating = None
+        if m_rating:
+            rating_value = float_or_none(m_rating.group('rating_value'))
+            best_rating = float_or_none(m_rating.group('best_rating'))
+            if rating_value and best_rating:
+                average_rating = rating_value / best_rating
```

Comment:
I wasn't sure if `average_rating` need to be scaled to 1 or not. If not we can simplify this to just `rating_value`